src/NodaTime/LocalDate.cs

```diff
@@ -285,6 +285,13 @@ public static LocalDate FromYearMonthWeekAndDay(int year, int month, int occurre
             return new LocalDate(year, month, targetDay);
         }
 
+        /// <summary>
+        /// Creates a <see cref="YearMonth"/> value for the month containing this date.
+        /// </summary>
+        /// <returns>A year/month value containing this date.</returns>
+        [Pure]
+        public YearMonth ToYearMonth() => new YearMonth(Year, Month, Calendar);
```

Comment:
Mmm. I think one reason to keep this is to avoid people just writing `new YearMonth(date.Year, date.Month)` which would be wrong in terms of any non-ISO calendar.

On the other hand, it's easier to add the feature later rather than remove it. I'm unsure :(